src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private ActionStrategy strategy = new ActionStrategyImpl();
+
+    @Override
+    public void generate(List<FruitTransaction> transactions) {
+        for (int i = 0; i < transactions.size(); i++) {
+            generateByRow(transactions.get(i));
+        }
+    }
+
+    void generateByRow(FruitTransaction transaction) {
+        strategy.get(transaction.getOperation())
```

Comment:
How does it work if `actionHandlerMap` field in `ActionStrategy` is not initialized?